core/cap-0046-10.md

```diff
@@ -135,100 +134,56 @@ The tricky scenario is 4, where a TB calls into a WB that calls into a mixture o
-
-We maintain a fork of Wasmi with metering added. This makes Wasmi is a tamed "wild component". 
-
-(Note this does not mean we are tied to a particular wasm implementation, it's just an example. If we decide to switch to a different interpreter or JIT in the future, we will be able to apply the same procedure to derive a new set of metering components.)
+This metering framework is generic and does not differentiate between the host and the WASM vm. Both the host and the vm are treated as components and blocks defined in the [specification](#specification) section and subject to the same metering procedures. 
 
+Our current choice of the WASM virtual machine implementation is Wasmi, which is a lightweight interpreter of the wasm standard, written in the same language (Rust) as the host. Wasmi runs an inner interpreter loop that executes a single wasm instruction on each loop. Thus every wasm instruction logic fits the requirements of a component. `WasmInsnExec` in `ContractCostType` is designated for the wasm instructions.
 
 ### Relation to cap-0046-07 (fee model)
 
-[CAP-0046-07](./cap-0046-07.md) proposed a fee model for smart contracts taking into account ledger access, storage and computation (or "gas"). This CAP details the computation aspect. However, this proposal identifies cpu and memory as separate aspects of the compute cost that needs to be budgeted separately. This difference needs to be resolved before this CAP finalize, i.e., either expand gas network settings in 07 or consolidate the `cpu_insns` and `mem_bytes` into a single "gas" parameter in here. 
+[CAP-0046-07](./cap-0046-07.md) proposed a fee model for smart contracts taking into account ledger access, storage and computation. This CAP details the computation aspect which includes cpu and memory. The metered `cpu_insns` goes into the fee model as input to the "compute" fee. While `mem_bytes` is not part of the fee model, it is subject to the network limit.
 
 ### Cost estimation
-This proposal relies on the "preflight" mechanism to provide users with cost estimation of a transaction. The total costs for each resource type as well as inputs to each individual cost type will be returned from the preflight simulation. These costs, however can only serve as guidance to the actual cost, since the ledger snapshot used for preflight may be outdated. Thus it is not guaranteed that a transaction staying below the budget during preflight will not exceed it during the actual run. 
+This proposal relies on the "preflight" mechanism to provide an estimation of the cpu and mem consumption in a transaction. These can only serve as guidance to the actual cost, since the ledger snapshot used for preflight may be outdated, as well as the actual logic during preflight and actual ("recording" vs "enforcing") modes maybe different. Thus it is not guaranteed that a transaction staying below the budget during preflight will not exceed it during the actual run. 
```

Comment:
```suggestion
This proposal relies on the "preflight" mechanism to provide an estimation of the cpu and mem consumption in a transaction. These can only serve as guidance to the actual cost, since the ledger snapshot used for preflight may be outdated, as well as the actual logic during preflight and actual ("recording" vs "enforcing") modes may be different. Thus it is not guaranteed that a transaction staying below the budget during preflight will not exceed it during the actual run. 
```